### DISCUSSION TITLE
What is the best way to add hundreds of thousands of records?

### DISCUSSION THREAD

**Discussion by oscadev** [pocketbase/pocketbase#1010] [Q&A]
I am trying to create each record in an array inside an async await for loop (per instructions on a similar question), but this is too slow. It has been about a day and it is at 300k out of 700k done. On something like postgres where I can import a CSV, it took me like 20 seconds to import millions of records. Is there an equivalent to this in Pocketbsae?

**Comment by ganigeorgiev** [comment]
Could you share your import code and how do you execute it?

You can always import large datasets directly using the sqlite3 cli tool, but you'll need to change your data format to the SQL INSERT, aka.:

```sql
INSERT INTO (id, created, updated, col1, col2, col3, col4)
VALUES
    ("test", "2022-01-01 00:00:00", "2022-01-01 00:00:00", ...),
    ...
```

(the sqlite3 cli took also supports the `.import` command allowing csv imports, but make sure that you have all required columns in place).

**Comment by vinceyoumans** [comment]
considering this is a Golang project, it would be very cool if there were some golang libraries that did this kind of thing.. perhaps a golang messaging service that listens for updates to the DB.  but I just started looking at this project today.
